component/site/views/account/view.html.php

```diff
@@ -48,164 +53,249 @@ class RedshopViewAccount extends RedshopView
 	 */
 	public function display($tpl = null)
 	{
-		/** @var JApplicationSite $app */
-		$app        = JFactory::getApplication();
-		$input      = $app->input;
-		$params     = $app->getParams('com_redshop');
-		$twigParams = [];
+		$this->params = $this->app->getParams('com_redshop');
+		$this->user   = JFactory::getUser();
+		$twigParams   = [
+			'url'    => \JURI::base(),
+			'app'    => $this->app,
+			'itemId' => $this->app->input->getInt('Itemid'),
+			'tagId'  => $this->app->input->getInt('tagid'),
+			'edit'   => $this->app->input->getInt('edit'),
+			'user'   => $this->user,
+			'params' => $this->params
+		];
 
 		RedshopHelperBreadcrumb::generate();
 
-		$itemId = $input->getInt('Itemid');
-		$layout = $input->getCmd('layout');
+		$itemId = $this->input->getInt('Itemid');
+		$layout = $this->input->getCmd('layout');
 
-		/** @var RedshopModelAccount $model */
-		$model = $this->getModel();
-		$user  = JFactory::getUser();
+		/** @var RedshopModelAccount $this */
+		$this->model = $this->getModel();
 
-		$userData = $model->getUserAccountInfo($user->id);
+		$this->userdata = $this->model->getUserAccountInfo($this->user->id);
 
-		if (empty($userData) && $layout != 'mywishlist') {
-			$app->redirect(
+		if (empty($this->userdata) && $layout != 'mywishlist') {
+			$this->app->redirect(
 				JRoute::_("index.php?option=com_redshop&view=account_billto&Itemid=" . $itemId),
 				JText::_('COM_REDSHOP_LOGIN_USER_IS_NOT_REDSHOP_USER')
 			);
 		}
 
-		$layout = $input->getCmd('layout', 'default');
-		$mail   = $input->getInt('mail');
+		$layout = $this->input->getCmd('layout', 'default');
+		$mail   = $this->input->getInt('mail');
 
 		// Preform security checks. Give permission to send wishlist while not logged in
-		if (($user->id == 0 && $layout != 'mywishlist') || ($user->id == 0 && $layout == 'mywishlist' && !isset($mail))) {
-			$app->redirect(JRoute::_('index.php?option=com_redshop&view=login&Itemid=' . $itemId, false));
+		if (($this->user->id == 0 && $layout != 'mywishlist') || ($this->user->id == 0 && $layout == 'mywishlist' && !isset($mail))) {
```

Comment:
@hungnguyenckc : just one advice for more security, using `===` that ensure type of input the same, some hack tips can be use base64 encode and fetch bad purpose, `==` is implicit convert then compare data with no care type of value.